### PR TITLE
MGMT-16479: Adding new command to lca that will run IBI installation flow

### PR DESCRIPTION
**Pull request by tsorya**
IBI FLOW: Adding new command to lca that will run IBI installation flow
Run state root preparation flow on iso
Precache images

### CODE REVIEW

File: ibu-imager/ops/execute.go [hunk 1]
```diff
@@ -67,7 +72,11 @@ func NewNsenterExecutor(logger *logrus.Logger, verbose bool) Execute {
 	return &nsenterExecutor{executor: executor{logger, verbose}}
 }
 
-func (e *nsenterExecutor) Execute(command string, args ...string) (string, error) {
+func (e *nsenterExecutor) ExecuteWithLiveLogger(command string, args ...string) (string, error) {
```
Comment (eranco74) [inline ibu-imager/ops/execute.go:75]: Something is off here.
you have `ExecuteWithLiveLogger` twice.
Also, why wrap `Execute` with `baseExecute`?

File: ibu-imager/ops/ops.go [hunk 2]
```diff
@@ -208,7 +208,7 @@ func (o *ops) ExtractTarWithSELinux(srcPath, destPath string) error {
 }
 
 func (o *ops) RemountSysroot() error {
-	_, err := o.hostCommandsExecutor.Execute("mount", "/sysroot", "-o", "remount,rw")
+	_, err := o.hostCommandsExecutor.Execute("mount", "/mnt/sysroot", "-o", "remount,rw")
```
Comment (eranco74) [inline ibu-imager/ops/ops.go:211]: Won't this break IBU?

File: internal/ostreeclient/ostreeclient.go [hunk 3]
```diff
@@ -15,41 +16,60 @@ type IClient interface {
...
 
 func (c *Client) Deploy(osname, refsepc string, kargs []string) error {
 	args := []string{"admin", "deploy", "--os", osname}
+	if c.ibi {
+		args = append(args, "--sysroot", "/mnt")
+	}
 	args = append(args, kargs...)
 	args = append(args, refsepc)
-	if c.IsOstreeAdminSetDefaultFeatureEnabled() {
+	if c.IsOstreeAdminSetDefaultFeatureEnabled() && !c.ibi {
```
Comment (eranco74) [inline internal/ostreeclient/ostreeclient.go:60]: Why not just set it as default in case of IBI?

File: ibu-imager/ops/ops.go (hunk 2 above)
Comment (tsorya) [reply ibu-imager/ops/ops.go:211]: mistake

File: internal/ostreeclient/ostreeclient.go (hunk 3 above)
Comment (tsorya) [reply internal/ostreeclient/ostreeclient.go:60]: I will change places between them but i don't fully understand the comment

File: ibu-imager/ops/execute.go (hunk 1 above)
Comment (tsorya) [reply ibu-imager/ops/execute.go:75]: cause this is interface with 2 functions? and here we have 3 different options of this interface. chroot, nsenter and os.cmd

File: main/precache-workload/main.go [hunk 4]
```diff
@@ -91,20 +89,18 @@ func main() {
 
 	log.Info("Loaded precache spec file.")
 
-	// Change root directory to /host
-	if err := syscall.Chroot(common.Host); err != nil {
-		terminateOnError(fmt.Errorf("failed to chroot to %s, err: %w", common.Host, err), Failure)
-	}
-	log.Infof("chroot %s successful", common.Host)
-
```
Comment (sakhoury) [inline main/precache-workload/main.go:99]: The pre-caching k8s job needs to chroot to the mounted host filesystem. May I ask why you removed this?

Comment (tsorya) [reply main/precache-workload/main.go:99]: I changed it to be the same as all other code. Now it uses chroot executor

**Pull request merged**